### PR TITLE
Add pip.conf workspace for pylint

### PR DESCRIPTION
**Pull request by GijsvanDulmen**
# Changes

Possibility to add an optional workspace to the pylint task in order to use pip.conf settings from a workspace.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations][authoring]
- [X] Includes [docs][docs] (if user facing)
- [X] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [X] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [X] Commit messages follow [commit message best practices][commit]
- [X] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/pylint/0.3/pylint.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        - name: HOME
+          value: /tmp/python
+        - name: PARAM_PIP_CONF_FILE
+          value: $(params.pip_conf_file)
+        - name: WORKSPACE_PIP_CONF_BOUND
+          value: $(workspaces.pip-conf.bound)
+        - name: WORKSPACE_PIP_CONF_PATH
+          value: $(workspaces.pip-conf.path)
+      script: |
+        export PATH=$PATH:$HOME.local/bin
```
Comment (vinamra28) [inline task/pylint/0.3/pylint.yaml:55]:
```suggestion
        export PATH=$PATH:$HOME/.local/bin
```

Comment (GijsvanDulmen) [reply task/pylint/0.3/pylint.yaml]: Resolved!

**Review by vinamra28 — APPROVED** [review]
/meow

**Comment by GijsvanDulmen** [comment]
> /test pull-catalog-catlin-lint

Added quotes to fix the last catlin output issues.

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**